cmd/create/cluster/cmd.go

```diff
@@ -981,80 +981,82 @@ func run(cmd *cobra.Command, _ []string) {
-		if len(billingAccounts) > 0 {
-			billingAccount, err = interactive.GetOption(interactive.Input{
-				Question: "Billing Account",
-				Help:     cmd.Flags().Lookup("billing-account").Usage,
-				Default:  billingAccount,
-				Required: true,
-				Options:  billingAccounts,
-			})
+		if interactive.Enabled() {
+			if len(billingAccounts) > 0 {
+				billingAccount, err = interactive.GetOption(interactive.Input{
+					Question: "Billing Account",
+					Help:     cmd.Flags().Lookup("billing-account").Usage,
+					Default:  billingAccount,
+					Required: true,
+					Options:  billingAccounts,
+				})
 
-			if err != nil {
-				r.Reporter.Errorf("Expected a valid billing account: '%s'", err)
-				os.Exit(1)
+				if err != nil {
+					r.Reporter.Errorf("Expected a valid billing account: '%s'", err)
+					os.Exit(1)
+				}
+
+				billingAccount = aws.ParseOption(billingAccount)
 			}
 
-			billingAccount = aws.ParseOption(billingAccount)
-		}
+			if billingAccount == "" || !ocm.IsValidAWSAccount(billingAccount) {
+				r.Reporter.Errorf("Expected a valid billing account")
+				os.Exit(1)
+			}
 
-		if billingAccount == "" || !ocm.IsValidAWSAccount(billingAccount) {
-			r.Reporter.Errorf("Expected a valid billing account")
-			os.Exit(1)
-		}
+			// Get contract info
+			contracts, isContractEnabled := GetBillingAccountContracts(cloudAccounts, billingAccount)
 
-		// Get contract info
-		contracts, isContractEnabled := GetBillingAccountContracts(cloudAccounts, billingAccount)
+			if billingAccount != awsCreator.AccountID {
+				r.Reporter.Infof("The selected AWS billing account is a different account than your AWS infrastructure account." +
+					"The AWS billing account will be charged for subscription usage. " +
+					"The AWS infrastructure account will be used for managing the cluster.")
+			} else {
+				r.Reporter.Infof("Using '%s' as billing account.",
+					billingAccount)
+			}
 
-		if billingAccount != awsCreator.AccountID {
-			r.Reporter.Infof("The selected AWS billing account is a different account than your AWS infrastructure account." +
-				"The AWS billing account will be charged for subscription usage. " +
-				"The AWS infrastructure account will be used for managing the cluster.")
-		} else {
-			r.Reporter.Infof("Using '%s' as billing account.",
-				billingAccount)
+			if isContractEnabled && len(contracts) > 0 {
+				//currently, an AWS account will have only one ROSA HCP active contract at a time
+				contractDisplay := GenerateContractDisplay(contracts[0])
+				r.Reporter.Infof(contractDisplay)
+			}
 		}
+	}
 
-		if isContractEnabled && len(contracts) > 0 {
-			//currently, an AWS account will have only one ROSA HCP active contract at a time
-			contractDisplay := GenerateContractDisplay(contracts[0])
-			r.Reporter.Infof(contractDisplay)
-		}
+	if isHostedCP && cmd.Flags().Changed("default-mp-labels") {
+		r.Reporter.Errorf("Setting the worker machine pool labels is not supported for hosted clusters")
+		os.Exit(1)
 	}
```

Comment:
We are checking `isHostedCP` above. Should this check go in the same block?

Reply:
Sure, let me move this in

Reply:
Updated